java-checks/src/main/java/org/sonar/java/checks/ReleaseSensorsCheck.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks;
+
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.IntStream;
+import org.sonar.check.Rule;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.JavaFileScannerContext;
+import org.sonar.plugins.java.api.semantic.MethodMatchers;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.NewClassTree;
+import org.sonar.plugins.java.api.tree.Tree;
+
+/**
+ * Rule S6889 flags issues when any sensor defined in {@link AcquireReleaseSensor} is not released.
+ * The intent of this rule is to make the developer aware of the need to release the sensor when it is not needed anymore.
+ * <p>
+ * The implemented logic checks that after acquiring a sensor there is at least one release invocation within the same file.
+ */
+@Rule(key = "S6889")
+public class ReleaseSensorsCheck extends IssuableSubscriptionVisitor {
+
+  enum AcquireReleaseSensor {
+    LOCATION_MANAGER("android.location.LocationManager", "requestLocationUpdates", "removeUpdates"),
+    SENSOR_MANAGER("android.hardware.SensorManager", "registerListener", "unregisterListener"),
+    CAMERA("android.hardware.Camera", "open", RELEASE),
```

Comment:
nice catch!
I had a look and in order to cover it we need a slightly different logic. Let's discuss it since the acquire is done on `CameraManager.openCamera()` and the release should be done invoking `CameraDevice.close()` in the callback provided to `CameraManager.openCamera()`